anime_downloader/util.py

```diff
@@ -268,25 +268,31 @@ def format_command(cmd, episode, file_format, speed_limit, path):
 
     cmd_dict = {
         '{aria2}': 'aria2c {stream_url} -x 12 -s 12 -j 12 -k 10M -o '
-                   '{file_format}.mp4 --continue=true --dir={download_dir}'
+                   '{file_format}{file_ext} --continue=true --dir={download_dir}'
                    ' --stream-piece-selector=inorder --min-split-size=5M --referer={referer} --check-certificate=false --user-agent={useragent} --max-overall-download-limit={speed_limit}',
         '{idm}'  : 'idman.exe /n /d {stream_url} /p {download_dir} /f {file_format}.mp4'
     }
 
+    file_name, file_ext = os.path.splitext(episode.source().stream_url)
```

Comment:
Get the final redirect url with requests.get(episode.source().stream_url, stream=True)